tuf_conformance/test_file_download.py

```diff
@@ -268,74 +310,93 @@ def test_multiple_changes_to_target(
-
+    # check file contents
     with open(client.get_last_downloaded_target(), "r") as last_download_file:
         assert file_contents.decode() == last_download_file.read()
-    
 
-    # Change the target contents 10 times and check it each time
+    # Now the repository makes 10 changes to the targets metadata.
+    # It modifies the targets file in the targets metadata including
+    # the hashes and length, and it also makes the corresponding
+    # content changes in the file itself. As such, the client
+    # should download the file
     for i in range(10):
+        # Create the file with the contents the repo will update to.
         new_file_contents = f"{file_contents.decode()}-{i}"
-        file_length = len(new_file_contents)
 
+        # Save the new file
         target_file = open(target_file_path, 'w+')
         target_file.write(new_file_contents)
         target_file.close()
 
+        # Bump targets in the repo
         repo.targets.version += 1
 
+        # Add a new target in the target repo
         target = TestTarget()
         target.path = target_base_name
         target.content = bytes(new_file_contents, 'utf-8')
-
         repo.add_target_with_length("targets", target)
+
+        # Bump repo snapshot
         repo.update_snapshot()
+
+        # Client updates
         client.refresh(init_data)
 
-        # Check that the file is the one we expect
+        # Before the client downloads, check
+        # that the fileis the one we expect
         assert os.path.isfile(target_file_path)
         with open(target_file_path) as f:
             assert f.read() == new_file_contents
-        target_file2 = client.download_target(init_data,
-                                              target_base_name,
-                                              target_base_url=url_prefix)
+
+        # The client now correctly downloads the new target file
+        client.download_target(init_data,
+                               target_base_name,
+                               target_base_url=url_prefix)
+
+        # Check that the client downloaded the file
         with open(client.get_last_downloaded_target(), "r") as last_download_file:
             assert last_download_file.read() == new_file_contents
 
-        # Substitute the file and check that the file is still the one we expect
-        # The client should not download this malicious one even though it exists
-        # at the same path as the legitimate one.
+        # The repository now creates a malicious file on with the
+        # same name a the legitimate file and bumps the targets
+        # and root metadata.
         malicious_file_contents_str = f"malicious-file-contents-{i}"
         new_target_file = open(target_file_path, 'w+')
         new_target_file.write(malicious_file_contents_str)
         new_target_file.close()
+        repo.targets.version += 1
+        repo.bump_root_by_one()
```

Comment:
these sort of changes seem a little strange:
* the root change affects nothing in the test as far as I can see
* the targets version change is a bit iffy: repository now serves a targets N+1 but it's not actually part of snapshot so likely no-one ever tries to download that -- the targets version is also bumped earlier in the loop (and snapshot is updated)